src/Components/CommentVote.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import { SquareArrowUp, SquareArrowDown } from "lucide-react";
+import { patchCommentByCommentId } from "../api";
+
+export default function CommentVote(props) {
+  const { comment } = props;
+
+  const storedPosVoteCount = localStorage.getItem(`commentPosVoteCount${comment.comment_id}`);
+  const storedNegVoteCount = localStorage.getItem(`commentNegVoteCount${comment.comment_id}`);
+
+  const [optimisticVotes, setOptimisticVotes] = useState(0);
+  const [commentPositiveVoteNumber, setCommentPositiveVoteNumber] = useState(storedPosVoteCount === "1" ? 1 : 0);
+
+  const [commentNegativeVoteNumber, setCommentNegativeVoteNumber] = useState(storedNegVoteCount === "1" ? 1 : 0);
+  const [commentError, setCommentError] = useState("");
+
+  function incrementVotes() {
```

Comment:
This is a neat way to make it like a user can only make one vote. If you truly want this behaviour it would be a good exercise in the future to extend your backend to enforce this